File: Safe/src/main/java/io/github/cfsima/modernsafe/util/VersionUtils.java
package io.github.cfsima.modernsafe.util;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import io.github.cfsima.modernsafe.R;

public class VersionUtils {
    public static String getVersionNumber(Context context) {
        try {
            PackageInfo pInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
            return pInfo.versionName;
        } catch (PackageManager.NameNotFoundException e) {
            return "Unknown";
        }
    }

    public static String getApplicationName(Context context) {
        return context.getString(R.string.app_name);
    }
}

File: Safe/src/main/java/io/github/cfsima/modernsafe/util/VersionUtils.kt
package io.github.cfsima.modernsafe.util

import android.content.Context
import android.content.pm.PackageManager
import io.github.cfsima.modernsafe.R

object VersionUtils {
    @JvmStatic
    fun getVersionNumber(context: Context): String {
        return try {
            val pInfo = context.packageManager.getPackageInfo(context.packageName, 0)
            pInfo.versionName ?: "Unknown"
        } catch (e: PackageManager.NameNotFoundException) {
            "Unknown"
        }
    }

    @JvmStatic
    fun getApplicationName(context: Context): String {
        return context.getString(R.string.app_name)
    }
}
